### DISCUSSION TITLE
Passing link function to `predict`

### DISCUSSION THREAD

**Discussion by pierreroudier** [inlabru-org/inlabru#144] [Ideas]
I have a model using a logit link function:

```R
mod_bru <- bru(
  components = 
    outcome ~ 
    -1 + 
    X(X, model = "fixed") + 
    psi(
      Z, 
      model = "iid",
      # hyper = priors,
      mapper = bru_mapper_matrix(labels = pred_names)
    ),
  family = "beta",
  data = my_data,
  options = list(
    control.family = list(link = 'logit'),
    control.predictor = list(link = 1, compute = TRUE)
  )
)
```

In order to back-transform my predictions, I find that I can't use `predict.bru` directly -- rather, I do:

```R
# Generate posterior samples
g <- generate(
  object = mod_bru,
  data = NULL,
  formula = ~ X_eval(my_data$X) + psi_eval(my_data$Z),
  n.samples = 100
)

# Apply back-transformation to the posteriors 
gg <- INLA::inla.link.invlogit(g) 

# Do some form of aggregation
res <- inlabru:::bru_summarise(gg)
```

Am I missing an easier/quicker way to do this? If not, I'd be happy to put together a PR for a `link = ...` option to `predict` (and passed to `generate`, with the default being the identy (`INLA::inla.link.identity`).

**Comment by finnlindgren** [comment]
You should be able to include the transformation in the prediction formula;
```
predict(
  object = mod_bru,
  data = NULL,
  formula = ~ plogis(X_eval(my_data$X) + psi_eval(my_data$Z)),
  n.samples = 100
)
```
(replace `plogis` with `INLA::inla.link.invlogit` if you specifically want to use the link function definition from `INLA`; it should be the same...)

Note: This should also work (it should first evaluate the component inputs, and then in the expression it will find and use the component effects instead when there are name clashes between data variables and component labels):
```
predict(
  object = mod_bru,
  data = my_data,
  formula = ~ plogis(X + Z),
  n.samples = 100
)
```

If the data is a list you likely need the development version of `inlabru` to avoid an error.
